planner/core/plan_cache.go

```diff
@@ -371,7 +371,7 @@ func rebuildRange(p Plan) error {
 					}
 				}
 				if pkCol != nil {
-					ranges, err := ranger.BuildTableRange(x.AccessConditions, x.ctx, pkCol.RetType)
+					ranges, _, _, err := ranger.BuildTableRange(x.AccessConditions, x.ctx, pkCol.RetType, 0)
```

Comment:
Add `todo` comment here and the following part.

Reply:
The logic is complete here and we don't need `TODO` here.

Firstly, we don't need to set range mem limit here(i.e., `rangeMaxSize` is set to 0). There are two reasons why we don't restrict range mem usage in `rebuildRange`.
1. The cached plan must be able to build complete ranges under mem limit when it is generated. Hence we can just build ranges from x.AccessConditions. The only difference between the last ranges and new ranges is the change of parameter values, which doesn't cause much change on the mem usage of complete ranges.
2. Different parameter values can change the mem usage of complete ranges. If we set range mem limit here, range fallback may heppen and cause correctness problem. For example, a in (?, ?, ?) is the access condition. When the plan is firstly generated, its complete ranges are ['a','a'], ['b','b'], ['c','c'], whose mem usage is under range mem limit 100B. When the cached plan is hit, the complete ranges may become ['aaa','aaa'], ['bbb','bbb'], ['ccc','ccc'], whose mem usage exceeds range mem limit 100B, and range fallback happens and tidb may fetch more rows than users expect.

Secondly, since there is no range mem limit, new `BuildTableRange` behaves just like old `BuildTableRange` and we can ignore the second and third return values(The second return value must be the same as `x.AccessConditions` and the third must be `nil`).

I add some notes on `rebuildRange`.